kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ClusterConfigsResourceIntegrationTest.java

```diff
@@ -0,0 +1,250 @@
+                + "/brokers/" + brokerId
+                + "/configs/compression.type")
+            .accept(Versions.JSON_API)
+            .get();
+    assertEquals(Status.OK.getStatusCode(), responseBrokerAfterUpdate.getStatus());
+
+    GetBrokerConfigResponse actualBrokerAfterUpdate =
+        responseBrokerAfterUpdate.readEntity(GetBrokerConfigResponse.class);
+    assertEquals(expectedBrokerAfterUpdate, actualBrokerAfterUpdate);
+
+    Response resetResponse =
+        request("/v3/clusters/" + clusterId + "/broker-configs/compression.type")
+            .accept(Versions.JSON_API)
+            .delete();
+    assertEquals(Status.NO_CONTENT.getStatusCode(), resetResponse.getStatus());
+
+    Response responseAfterReset =
+        request("/v3/clusters/" + clusterId + "/broker-configs/compression.type")
+            .accept(Versions.JSON_API)
+            .get();
+    assertEquals(Status.NOT_FOUND.getStatusCode(), responseAfterReset.getStatus());
+
+    GetBrokerConfigResponse expectedBrokerAfterReset =
+        new GetBrokerConfigResponse(
+            new BrokerConfigData(
+                "crn:///kafka=" + clusterId
+                    + "/broker=" + brokerId
+                    + "/config=compression.type",
+                new ResourceLink(
+                    baseUrl
+                        + "/v3/clusters/" + clusterId
+                        + "/brokers/" + brokerId
+                        + "/configs/compression.type"),
+                clusterId,
+                brokerId,
+                "compression.type",
+                "producer",
+                /* isDefault= */ true,
+                /* isReadOnly= */ false,
+                /* isSensitive= */ false,
+                ConfigSource.DEFAULT_CONFIG,
+                singletonList(
+                    new ConfigSynonymData(
+                        "compression.type", "producer", ConfigSource.DEFAULT_CONFIG))));
+
+    Response brokerResponseAfterReset =
+        request(
+            "/v3/clusters/" + clusterId
+                + "/brokers/" + brokerId
+                + "/configs/compression.type")
+            .accept(Versions.JSON_API)
+            .get();
+    assertEquals(Status.OK.getStatusCode(), brokerResponseAfterReset.getStatus());
+
+    GetBrokerConfigResponse actualBrokerAfterReset =
+        brokerResponseAfterReset.readEntity(GetBrokerConfigResponse.class);
+    assertEquals(expectedBrokerAfterReset, actualBrokerAfterReset);
+  }
+
+  @Test
+  public void updateClusterConfig_nonExistingCluster_throwsNotFound() {
+    Response response =
+        request("/v3/clusters/foobar/broker-configs/compression.type")
+            .accept(Versions.JSON_API)
+            .put(
+                Entity.entity(
+                    "{\"data\":{\"attributes\":{\"value\":\"producer\"}}}", Versions.JSON_API));
+    assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+  }
+
+  @Test
+  public void resetClusterConfig_nonExistingCluster_throwsNotFound() {
```

Comment:
How about resetting  cluster config for a non existing config ?

Reply:
As explained in the other comment, we can't fail if the config does not exist. It's kind of a lame behaviour which if I have time I will try to fix in AK at some point.